packages/gatsby-plugin-google-tagmanager/src/gatsby-browser.js

```diff
@@ -0,0 +1,6 @@
+exports.onRouteUpdate = ({ location }) => {
+  // wrap inside a timeout to ensure the title has properly been changed
+  setTimeout(() => {
+    window.dataLayer.push({ event: `gatsby-route-change` })
+  }, 2000)
```

Comment:
That's quite big timeout IMO, requesting animiation frame or `0`s timeout should work here too.

Also we should keep track of timeout and cancel it if another `onRouteUpdate` call was triggered to avoid pushing `gatsby-route-change` event twice for same path

Reply:
In what case is another onRouteUpdate triggered? I would think that if someone visits one page, then quickly visits another, we should log both page views.

Reply:
Right, but this object you are pushing doesn't include any page metadata (just event name), so I assumed it's GTM services that extracts information on events. So if something like this would happen:

 - navigate to `/page-A`, (start setTimeout)
 - quickly (< 2s) navigate to `/page-B` (start another setTimeout)
 - first timeout callback is exucuted (push event when on `/page-B`)
 - second timeout callback is exucuted (push event when on `/page-B`)

In any case - I think 2000 is just too long here. If you are using `react-helmet` to handle `<title>` etc, it should be enough to use `setTimeout(<some-code>, 0)`

Reply:
Got it, thanks for the insight. I drastically lowered the delay -- do you still think it's necessary to keep a reference to the timeout to cancel it?